bus-mapping/src/evm/opcodes.rs

```diff
@@ -345,15 +345,20 @@ pub fn gen_begin_tx_ops(state: &mut CircuitInputStateRef) -> Result<ExecStep, Er
         state.call_context_write(&mut exec_step, call.call_id, field, value);
     }
 
-    // Increase caller's nonce
     let caller_address = call.caller_address;
-    let nonce_prev = state.sdb.get_account(&caller_address).1.nonce;
+    let nonce_prev = state.sdb.get_account(&caller_address).1.nonce.as_u64();
+    // Increase caller's nonce when the tx is not invalid
+    let nonce = if !state.tx.invalid_tx {
+        state.sdb.increase_nonce(&caller_address) + 1
```

Comment:
I think this line should be ```nonce_prev+1```

Reply:
If you look at the implementation I think that's what this does, because `increase_nonce`, a bit surprisingly, returns the old nonce, not the new nonce.